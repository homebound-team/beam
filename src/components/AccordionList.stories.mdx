import { Meta, Canvas, Story, ArgsTable } from "@storybook/addon-docs";
import { AccordionList } from "./AccordionList";

<Meta
  title="Catalog/AccordionList"
  component={AccordionList}
  args={{
    accordions: [
      {
        title: "Iron Man",
        children:
          "Genius. Billionaire. Philanthropist. Tony Stark's confidence is only matched by his high-flying abilities as the hero called Iron Man.",
      },
      {
        title: "Captain Marvel",
        children:
          "Carol Danvers becomes one of the universe's most powerful heroes when Earth is caught in the middle of a galactic war between two alien races.",
      },
      {
        title: "Captain America",
        children:
          "Recipient of the Super Soldier serum, World War II hero Steve Rogers fights for American ideals as one of the worldâ€™s mightiest heroes and the leader of the Avengers.",
      },
      {
        title: "Thor",
        children:
          "The son of Odin uses his mighty abilities as the God of Thunder to protect his home Asgard and planet Earth alike.",
      },
      {
        title: "Black Widow",
        children:
          "Natasha Romanoff, separated from the now-fractured Avengers, confronts the dark path she took to becoming a spy and assassin, as well as events that followed.",
      },
    ],
  }}
  argTypes={{
    accordions: { table: { control: { disable: true } } },
    allowMultipleExpanded: { table: { defaultValue: { summary: true } } },
    size: {
      table: { defaultValue: { summary: "undefined" } },
      control: { type: "select" },
      options: [undefined, "xs", "sm", "md", "lg"],
    },
  }}
/>

export const Template = (args) => {
  return <AccordionList {...args} />;
};

# Accordion

<Canvas>
  <Story name="AccordionList">{Template.bind({})}</Story>
</Canvas>

<ArgsTable story="AccordionList" />
