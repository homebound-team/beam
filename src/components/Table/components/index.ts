export { defaultRenderFn, headerRenderFn, rowClickRenderFn, rowLinkRenderFn } from "./cell";
export type { GridCellContent, RenderCellFn } from "./cell";
export * from "./CollapseToggle";
export { EditColumnsButton } from "./EditColumnsButton";
export { Row } from "./Row";
export * from "./SelectToggle";
export * from "./SortHeader";
