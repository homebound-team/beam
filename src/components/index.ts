export * from "./Button";
export * from "./ButtonGroup";
export * from "./Chip";
export * from "./CssReset";
export * from "./Filters";
export * from "./GridTable";
export * from "./Icon";
export * from "./IconButton";
export * from "./Modal/Modal";
export * from "./Modal/ModalContext";
export { NavLink } from "./NavLink";
export * from "./Pill";
export * from "./RichTextField";
export * from "./SuperDrawer";
export { TabsWithContent } from "./Tabs";
export * from "./Tooltip";
