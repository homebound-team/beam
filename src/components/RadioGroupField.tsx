import { ReactNode, useMemo, useRef } from "react";
import { useRadio, useRadioGroup } from "react-aria";
import { RadioGroupState, useRadioGroupState } from "react-stately";
import { Css } from "src/Css";

let nextNameId = 0;

export interface RadioFieldOption<K extends string> {
  // testId?: string;
  /** The label for a specific option, i.e. "Cheddar". */
  label: string;
  /** An optional longer description to render under the label. */
  description?: string | (() => ReactNode);
  /** The undisplayed value, i.e. an id of some sort. */
  value: K;
}

interface RadioGroupFieldProps<K extends string> {
  // id?: string;
  /** The label for the choice itself, i.e. "Favorite Cheese". */
  label: string;
  /** The currently selected option value (i.e. an id). */
  value: K | undefined;
  /** Called when an option is selected. We don't support unselecting. */
  onChange: (value: K) => void;
  /** The list of options. */
  options: RadioFieldOption<K>[];
  disabled?: boolean;
}

/**
 * Provides a radio group with label.
 *
 * This is generally meant to be used in a form vs. being raw radio buttons.
 *
 * TODO: Add hover (non selected and selected) styles
 */
export function RadioGroupField<K extends string>(props: RadioGroupFieldProps<K>) {
  const { label, value, onChange, options, disabled = false } = props;

  // useRadioGroupState uses a random group name, so use our name
  const name = useMemo(() => `radio-group-${++nextNameId}`, []);
  const state = useRadioGroupState({
    name,
    value,
    onChange: (value) => onChange(value as K),
    isDisabled: disabled,
    isReadOnly: false,
  });

  // We use useRadioGroup b/c it does neat keyboard up/down stuff
  // TOOD: Pass read only, required, error message to useRadioGroup
  const { labelProps, radioGroupProps } = useRadioGroup({ label, isDisabled: disabled }, state);

  // max-width is dependent on having descriptions
  const anyDescriptions = options.some((o) => !!o.description);

  return (
    <div css={Css.maxw(anyDescriptions ? "344px" : "320px").$}>
      <div css={Css.sm.gray800.my1.$} {...labelProps}>
        {label}
      </div>
      <div {...radioGroupProps}>
        {options.map((option) => (
          <Radio key={option.value} parentId={state.name} option={option} state={state} />
        ))}
      </div>
    </div>
  );
}

// Not meant to be standalone, but its own component so it can use hooks
function Radio<K extends string>(props: { parentId: string; option: RadioFieldOption<K>; state: RadioGroupState }) {
  const {
    parentId,
    option: { description, label, value },
    state,
  } = props;
  const disabled = state.isDisabled;

  const labelId = `${parentId}-${value}-label`;
  const descriptionId = `${parentId}-${value}-description`;
  const ref = useRef<HTMLInputElement>(null);
  const { inputProps } = useRadio({ value, "aria-labelledby": labelId }, state, ref);

  return (
    <label css={Css.df.cursorPointer.mb1.if(disabled).add("cursor", "initial").$}>
      <input
        type="radio"
        ref={ref}
        css={{
          ...radioReset,
          ...(!disabled && state.selectedValue === value ? radioChecked : radioUnchecked),
          ...(disabled ? radioDisabled : {}),
          // Sometimes we use useFocusRing, but here we want both mouse & keyboard to drive focus
          ...{ "&:focus": radioFocus },
          // Nudge down so the center of the circle lines up with the label text
          ...Css.mtPx(2).mr1.$,
        }}
        disabled={disabled}
        {...inputProps}
        aria-labelledby={labelId}
      />
      <div>
        <div
          id={labelId}
          css={Css.smEm.gray800.if(disabled).gray400.$}
          {...(description ? { "aria-describedby": descriptionId } : {})}
        >
          {label}
        </div>
        {description && (
          <div id={descriptionId} css={Css.sm.gray700.if(disabled).gray400.$}>
            {description}
          </div>
        )}
      </div>
    </label>
  );
}

const whiteCircle =
  "data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='white' xmlns='http://www.w3.org/2000/svg'%3e%3ccircle cx='8' cy='8' r='4'/%3e%3c/svg%3e";

// Didn't put these in radioReset yet, are they needed?
// color-adjust: exact;
// -webkit-print-color-adjust: exact;

export const radioReset = {
  ...Css.add("appearance", "none").p0.dib.vMid.add("userSelect", "none").fs0.h2.w2.br100.$,
  ...Css.add("outline", "0px solid transparent").$,
  // By default we're white with a blue border
  ...Css.bgWhite.bLightBlue900.ba.$,
  // Set a color that will be used by background=currentColor + box shadow, but is initially ignored
  ...Css.lightBlue700.$,
};

// Unchecked means a gray border
export const radioUnchecked = Css.cursorPointer.bGray300.$;

// Checked means a blue circle (achieved by a blue background + white dot background image)
export const radioChecked = {
  // When checked, make a circle by having the border blend into the background
  ...Css.bTransparent.$,
  // And make the background become the current (blue) color
  ...Css.add("backgroundColor", "currentColor")
    .add("backgroundSize", "100% 100%")
    .add("backgroundPosition", "center")
    .add("backgroundRepeat", "no-repeat").$,
  // And use backgroundImage to draw a white dot in the middle of the background
  ...Css.add("backgroundImage", `url("${whiteCircle}")`).$,
};

// When active draw another circle via boxShadow
export const radioFocus = {
  // The box shadow ends up drawing over this afaict?
  ...Css.add("outline", "2px solid transparent").add("outlineOffset", "2px").$,
  // Draw 1st box shadow of white/outline, 2nd box current (blue) of another outline
  ...Css.bshFocus.$,
};

export const radioDisabled = {
  ...Css.cursorNotAllowed.gray100.$,
  ...Css.add("backgroundColor", "currentColor")
    .add("backgroundSize", "100% 100%")
    .add("backgroundPosition", "center")
    .add("backgroundRepeat", "no-repeat").$,
};
