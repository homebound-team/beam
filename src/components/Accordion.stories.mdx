import { Meta, Canvas, Story, ArgsTable } from "@storybook/addon-docs";
import { Accordion } from "./Accordion";

<Meta
  title="Catalog/Accordion"
  component={Accordion}
  args={{ title: "Captain America" }}
  argTypes={{
    title: { control: { type: "text" }, table: { defaultValue: { summary: undefined } } },
    topBorder: { table: { defaultValue: { summary: true } } },
    disabled: { table: { defaultValue: { summary: false } } },
    bottomBorder: { table: { defaultValue: { summary: false } } },
    defaultExpanded: { table: { defaultValue: { summary: false } } },
    size: {
      table: { defaultValue: { summary: "undefined" } },
      control: { type: "select" },
      options: [undefined, "xs", "sm", "md", "lg"],
    },
  }}
/>

export const Template = (args) => {
  return (
    <Accordion {...args}>
      Recipient of the Super Soldier serum, World War II hero Steve Rogers fights for American ideals as one of the
      worldâ€™s mightiest heroes and the leader of the Avengers.
    </Accordion>
  );
};

# Accordion

<Canvas>
  <Story name="Accordion">{Template.bind({})}</Story>
</Canvas>

<ArgsTable story="Accordion" exclude={["index", "setExpandedIndex"]} />
