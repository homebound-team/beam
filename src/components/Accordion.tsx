import { useId, useResizeObserver } from "@react-aria/utils";
import { Dispatch, ReactNode, SetStateAction, useCallback, useEffect, useMemo, useRef, useState } from "react";
import { useFocusRing } from "react-aria";
import { Icon } from "src/components/Icon";
import { Css, Only, Padding, Palette, Xss } from "src/Css";
import { useTestIds } from "src/utils";

type AccordionXss = Xss<Padding>;

export interface AccordionProps<X = AccordionXss> {
  title: ReactNode;
  children: ReactNode;
  disabled?: boolean;
  defaultExpanded?: boolean;
  size?: AccordionSize;
  /** Adds a top border (enabled by default) */
  topBorder?: boolean;
  /** Adds a bottom border (disabled by default) */
  bottomBorder?: boolean;
  /**
   * Used by AccordionList
   * Allows multiple accordions to be expanded simultaneously (enabled by default)
   */
  index?: number;
  setExpandedIndex?: Dispatch<SetStateAction<number | undefined>>;
  /** Turns the title into a button. If provided, disables expand/collapse on title text */
  titleOnClick?: VoidFunction;
  /** Used by Accordion list. Sets default padding to 0 for nested accordions */
  omitPadding?: boolean;
  /** Styles overrides for padding */
  xss?: X;
  /** Modifies the typography, padding, icon size and background color of the accordion header */
  compact?: boolean;
}

export function Accordion<X extends Only<AccordionXss, X>>(props: AccordionProps<X>) {
  const {
    title,
    children,
    size,
    disabled = false,
    defaultExpanded = false,
    compact = false,
    topBorder = compact ? false : true,
    bottomBorder = false,
    index,
    setExpandedIndex,
    titleOnClick,
    omitPadding = false,
    xss,
  } = props;
  const tid = useTestIds(props, "accordion");
  const id = useId();
  const [expanded, setExpanded] = useState(defaultExpanded && !disabled);
  const { isFocusVisible, focusProps } = useFocusRing();
  const contentRef = useRef<HTMLDivElement>(null);
  // On initial render, if the accordion is expanded, then set `height` to auto to avoid unnecessary animation on render.
  const [contentHeight, setContentHeight] = useState(expanded ? "auto" : "0");

  useEffect(() => {
    setExpanded(defaultExpanded && !disabled);
  }, [defaultExpanded, disabled]);

  useEffect(() => {
    // When the `expanded` value changes - If true, it means the Accordion's content has been rendered, Otherwise, it's been hidden
    // Then when the content is displayed, the calculate its height so we can give this value to the container to animate height smoothly.
    // When content is removed, simply set the height back to 0
    setContentHeight(expanded && contentRef.current ? `${contentRef.current.scrollHeight}px` : "0");
  }, [expanded]);

  // Using a resizing observer to check if the content of the accordion changes (i.e. lazy loaded image, auto-sizing textarea, etc..),
  // If it does change, then we need to update the container's height accordingly. Only update the height if the accordion is expanded.
  // Note - This may result in two `setContentHeight` calls when the accordion opens: (1) via the above `useEffect` and (2) in `onResize`
  //        Both `setContentHeight` calls _should_ set the same value, so no unnecessary re-renders would be triggered, making this a harmless additional set call.
  const onResize = useCallback(() => {
    if (contentRef.current && expanded) {
      setContentHeight(`${contentRef.current.scrollHeight}px`);
    }
  }, [expanded, setContentHeight]);
  useResizeObserver({ ref: contentRef, onResize });

  const toggle = useCallback(() => {
    setExpanded((prev) => !prev);
    if (setExpandedIndex) setExpandedIndex(index);
  }, [index, setExpandedIndex]);

  const touchableStyle = useMemo(
    () => ({
      ...Css.df.jcsb.gapPx(12).aic.p2.baseMd.outline("none").onHover.bgGray100.if(!!titleOnClick).baseSb.$,
      ...(compact && Css.smMd.pl2.prPx(10).py1.bgGray100.mbPx(4).br8.onHover.bgGray200.$),
      ...(compact && !!titleOnClick && Css.br0.$),
      ...(disabled && Css.gray500.$),
      ...(isFocusVisible && Css.boxShadow(`inset 0 0 0 2px ${Palette.Blue700}`).$),
      ...xss,
    }),
    [compact, disabled, isFocusVisible, titleOnClick, xss],
  );

  return (
    <div
      {...tid.container}
      css={{
        ...Css.bcGray300.if(topBorder).bt.if(bottomBorder).bb.$,
        ...(size ? Css.wPx(accordionSizes[size]).$ : {}),
      }}
    >
      {titleOnClick ? (
        <div {...focusProps} aria-controls={id} aria-expanded={expanded} css={Css.df.$}>
          <button {...tid.title} disabled={disabled} css={{ ...touchableStyle, ...Css.fg1.$ }} onClick={titleOnClick}>
            {title}
          </button>
          <button
            {...tid.toggle}
            disabled={disabled}
            css={{ ...touchableStyle, ...Css.px2.jcfe.if(compact).pxPx(10).$ }}
            onClick={toggle}
          >
            <RotatingChevronIcon expanded={expanded} />
          </button>
        </div>
      ) : (
        <button
          {...tid.title}
          {...focusProps}
          aria-controls={id}
          aria-expanded={expanded}
          disabled={disabled}
          css={{ ...Css.w100.$, ...touchableStyle }}
          onClick={toggle}
        >
          <span css={Css.fg1.tal.$}>{title}</span>
          <RotatingChevronIcon expanded={expanded} />
        </button>
      )}
      <div
        {...tid.details}
        id={id}
        aria-hidden={!expanded}
        css={Css.oh.h(contentHeight).add("transition", "height 250ms ease-in-out").$}
      >
        {expanded && (
          <div css={Css.px2.pb2.pt1.if(omitPadding).p0.$} ref={contentRef} {...tid.content}>
            {children}
          </div>
        )}
      </div>
    </div>
  );
}

function RotatingChevronIcon(props: { expanded: boolean }) {
  return (
    <span
      css={{
        ...Css.fs0.$,
        transition: "transform 250ms linear",
        transform: props.expanded ? "rotate(180deg)" : "rotate(0deg)",
      }}
    >
      <Icon icon="chevronDown" />
    </span>
  );
}

export type AccordionSize = "xs" | "sm" | "md" | "lg";

const accordionSizes: Record<AccordionSize, number> = {
  xs: 240,
  sm: 360,
  md: 480,
  lg: 600,
};
