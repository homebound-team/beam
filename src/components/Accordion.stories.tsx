import { Meta } from "@storybook/react";
import { Css } from "src/Css";
import { Accordion } from "./Accordion";

export default {
  component: Accordion,
  title: "Workspace/Components/Accordion",
  parameters: { backgrounds: { default: "white" } },
} as Meta;

export function AccordionVariations() {
  return (
    <>
      <Accordion title="Top border disabled" topBorder={false}>
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
      <Accordion title="Default expanded" defaultExpanded>
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
      <Accordion title="Disabled" disabled>
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
      <Accordion title="Bottom border enabled" bottomBorder>
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
    </>
  );
}
export function AccordionSizes() {
  return (
    <>
      <Accordion title="Extra small" size="xs">
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
      <Accordion title="Small" size="sm">
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
      <Accordion title="Medium" size="md">
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
      <Accordion title="Large" size="lg">
        <div css={Css.sm.$}>
          Our modern approach to homebuilding makes the process easier and more personal than ever before.
        </div>
      </Accordion>
    </>
  );
}
