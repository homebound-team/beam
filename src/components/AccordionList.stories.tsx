import { Meta } from "@storybook/react";
import { AccordionProps } from "./Accordion";
import { AccordionList } from "./AccordionList";

export default {
  component: AccordionList,
  title: "Workspace/Components/Accordion",
  parameters: { backgrounds: { default: "white" } },
} as Meta;

export function AccordionListWithMultipleSelections() {
  const accordions: AccordionProps[] = [
    { title: "First accordion title", children: <div>Fisrt accordion description</div> },
    { title: "Second accordion title", children: <div>Second accordion description</div> },
    { title: "Third accordion title", children: <div>Third accordion description</div> },
  ];
  return <AccordionList accordions={accordions} />;
}

export function AccordionListWithOneSelection() {
  const accordions: AccordionProps[] = [
    { title: "First accordion title", children: <div>Fisrt accordion description</div> },
    { title: "Second accordion title", children: <div>Second accordion description</div>, defaultExpanded: true },
    { title: "Third accordion title", children: <div>Third accordion description</div> },
  ];
  return <AccordionList accordions={accordions} allowMultipleExpanded={false} />;
}
