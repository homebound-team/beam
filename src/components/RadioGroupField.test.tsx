import { click, render } from "@homebound/rtl-utils";
import { RadioGroupField } from "src/components/RadioGroupField";

describe("RadioGroupField", () => {
  it("renders with just labels", async () => {
    const r = await render(
      <RadioGroupField
        label={"Favorite cheese"}
        value="a"
        onChange={() => {}}
        options={[
          { label: "Asiago", value: "a" },
          { label: "Burratta", value: "b" },
        ]}
      />,
    );
    expect(r.baseElement).toMatchSnapshot();
  });

  it("renders with just labels and description", async () => {
    const r = await render(
      <RadioGroupField
        label={"Favorite cheese"}
        value="a"
        onChange={() => {}}
        options={[
          { value: "a", label: "Asiago" },
          {
            value: "b",
            label: "Burratta",
            description: "Burrata is an Italian cow milk cheese made from mozzarella and cream.",
          },
        ]}
      />,
    );
    expect(r.baseElement).toMatchSnapshot();
  });

  it("has data-testids for its options", async () => {
    const r = await render(
      <RadioGroupField
        label={"Favorite cheese"}
        value="a"
        onChange={() => {}}
        options={[
          { value: "a", label: "Asiago" },
          { value: "b", label: "Burratta" },
        ]}
      />,
    );
    click(r.favoriteCheese_a);
  });
});
