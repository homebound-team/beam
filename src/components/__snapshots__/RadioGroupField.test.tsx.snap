// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroupField renders with just labels 1`] = `
.emotion-0 {
  max-width: 320px;
}

.emotion-1 {
  color: rgba(100,100,100,1);
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  margin-bottom: 4px;
}

.emotion-2 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
}

.emotion-3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: currentColor;
  border-color: currentColor;
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-background-size: 100% 100%;
  background-size: 100% 100%;
  background-image: url("data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='white' xmlns='http://www.w3.org/2000/svg'%3e%3ccircle cx='8' cy='8' r='4'/%3e%3c/svg%3e");
  margin-right: 8px;
  margin-top: 2px;
}

.emotion-4 {
  color: rgba(53,53,53,1);
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
}

.emotion-6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: rgba(254,254,254,1);
  border-color: rgba(221,220,220,1);
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  cursor: pointer;
  margin-right: 8px;
  margin-top: 2px;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <label
        class="emotion-1"
        data-testid="radio_label"
        id="react-aria-2"
      >
        Favorite cheese
      </label>
      <div
        aria-labelledby="react-aria-2"
        aria-orientation="vertical"
        id="react-aria-1"
        role="radiogroup"
      >
        <label
          class="emotion-2"
        >
          <input
            aria-labelledby="radio-group-1-a-label"
            checked=""
            class="emotion-3"
            name="react-aria-3"
            tabindex="0"
            type="radio"
            value="a"
          />
          <div>
            <div
              class="emotion-4"
              id="radio-group-1-a-label"
            >
              Asiago
            </div>
          </div>
        </label>
        <label
          class="emotion-2"
        >
          <input
            aria-labelledby="radio-group-1-b-label"
            class="emotion-6"
            name="react-aria-3"
            tabindex="0"
            type="radio"
            value="b"
          />
          <div>
            <div
              class="emotion-4"
              id="radio-group-1-b-label"
            >
              Burratta
            </div>
          </div>
        </label>
      </div>
    </div>
  </div>
</body>
`;

exports[`RadioGroupField renders with just labels and description 1`] = `
.emotion-0 {
  max-width: 344px;
}

.emotion-1 {
  color: rgba(100,100,100,1);
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  margin-bottom: 4px;
}

.emotion-2 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
}

.emotion-3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: currentColor;
  border-color: currentColor;
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-background-size: 100% 100%;
  background-size: 100% 100%;
  background-image: url("data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='white' xmlns='http://www.w3.org/2000/svg'%3e%3ccircle cx='8' cy='8' r='4'/%3e%3c/svg%3e");
  margin-right: 8px;
  margin-top: 2px;
}

.emotion-4 {
  color: rgba(53,53,53,1);
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
}

.emotion-6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: rgba(254,254,254,1);
  border-color: rgba(221,220,220,1);
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms;
  cursor: pointer;
  margin-right: 8px;
  margin-top: 2px;
}

.emotion-8 {
  color: rgba(100,100,100,1);
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <label
        class="emotion-1"
        data-testid="radio_label"
        id="react-aria-5"
      >
        Favorite cheese
      </label>
      <div
        aria-labelledby="react-aria-5"
        aria-orientation="vertical"
        id="react-aria-4"
        role="radiogroup"
      >
        <label
          class="emotion-2"
        >
          <input
            aria-labelledby="radio-group-2-a-label"
            checked=""
            class="emotion-3"
            name="react-aria-6"
            tabindex="0"
            type="radio"
            value="a"
          />
          <div>
            <div
              class="emotion-4"
              id="radio-group-2-a-label"
            >
              Asiago
            </div>
          </div>
        </label>
        <label
          class="emotion-2"
        >
          <input
            aria-labelledby="radio-group-2-b-label"
            class="emotion-6"
            name="react-aria-6"
            tabindex="0"
            type="radio"
            value="b"
          />
          <div>
            <div
              aria-describedby="radio-group-2-b-description"
              class="emotion-4"
              id="radio-group-2-b-label"
            >
              Burratta
            </div>
            <div
              class="emotion-8"
              id="radio-group-2-b-description"
            >
              Burrata is an Italian cow milk cheese made from mozzarella and cream.
            </div>
          </div>
        </label>
      </div>
    </div>
  </div>
</body>
`;
